File: app/src/main/java/me/imzack/app/ender/view/fragment/BaseListFragment.java
package me.imzack.app.ender.view.fragment;

public abstract class BaseListFragment extends BaseFragment {

    private OnListScrolledListener mOnListScrolledListener;

    public interface OnListScrolledListener {
        void onListScrolled(int variation);
    }

    public void setOnListScrolledListener(OnListScrolledListener listener) {
        mOnListScrolledListener = listener;
    }

    public void onListScrolled(int variation) {
        if (mOnListScrolledListener != null) {
            mOnListScrolledListener.onListScrolled(variation);
        }
    }
}

File: app/src/main/java/net/zackzhang/app/end/view/fragment/BaseListFragment.kt
package net.zackzhang.app.end.view.fragment

abstract class BaseListFragment : BaseFragment() {

    var mOnListScrolledListener: ((variation: Int) -> Unit)? = null

    fun onListScrolled(variation: Int) {
        mOnListScrolledListener?.invoke(variation)
    }
}
